File: retrofit/src/main/java/com/kcst/retrofit/RequestHandler.java
package com.kcst.retrofit;

import com.google.gson.Gson;

import org.json.JSONException;

import java.io.IOException;

public interface RequestHandler {
    <T> Result<T> requestHandler(RetrofitService retrofitService, BaseRequest baseRequest, Class<T> baseResponse, Gson gson);
}

File: retrofit/src/main/java/com/kcst/retrofit/RequestHandler.kt
package com.kcst.retrofit

import com.google.gson.Gson
import java.lang.reflect.InvocationTargetException

interface RequestHandler {
    suspend fun <T : BaseResponse<*>> requestHandler(
        retrofitService: RetrofitService,
        baseRequest: BaseRequest,
        responseClazz: Class<T>,
    ): T
}